sonic-iOS/SonicSample/SonicSample/SonicJSContext.m

```diff
@@ -24,20 +24,19 @@ @implementation SonicJSContext
 
 - (void)getDiffData:(NSDictionary *)option withCallBack:(JSValue *)jscallback
 {
-    JSValue *callback = self.owner.jscontext.globalObject;
+    SonicWebViewController *owner = self.owner;
+    if (!owner) { return; }
     
-    __weak typeof(self) weakSelf = self;
-    [[SonicEngine sharedEngine] sonicUpdateDiffDataByWebDelegate:self.owner completion:^(NSDictionary *result) {
-       
-        if (result) {
-            
+    __weak typeof(SonicWebViewController *) weakOwner = owner;
+    [[SonicEngine sharedEngine] sonicUpdateDiffDataByWebDelegate:owner completion:^(NSDictionary *result) {
+        __strong typeof(SonicWebViewController *) strongOwner = weakOwner;
+        if (strongOwner && result) {
+            JSValue *callback = strongOwner.jscontext.globalObject;
             NSData *json = [NSJSONSerialization dataWithJSONObject:result options:NSJSONWritingPrettyPrinted error:nil];
             NSString *jsonStr = [[NSString alloc]initWithData:json encoding:NSUTF8StringEncoding];
             
             [callback invokeMethod:@"getDiffDataCallback" withArguments:@[jsonStr]];
         }
-        //fix:https://github.com/Tencent/VasSonic/issues/251
-        weakSelf.owner = nil;
```

Comment:
We can't just set `nil` in here, it would `window.sonic.getPerformance()` invalid and any other intercommunication.